src/pages/content-edit/head.js

```diff
@@ -3,261 +3,431 @@ import format from "date-fns/format";
+
+        updated = {
             // TODO: Remove `published` field, it's deprecated
-            published      : parseInt(format(start, "x"), 10),
-            published_at   : parseInt(format(start, "x"), 10),
-            published_by   : user,
-            unpublished_at : end ? parseInt(format(end, "x"), 10) : null,
-            unpublished_by : end ? user : null
+            published    : startTs,
+            published_at : startTs,
+            published_by : user
+        };
+
+        if(ctrl.end.ts && (ctrl.start.ts > ctrl.end.ts)) {
+            ctrl.end = ctrl.nulledScheduleEnd();
+
+            updated.unpublished_at = null;
+            updated.unpublished_by = null;
+
+            console.warn("Invalid end date. Resetting end date.");
+        }
+
+        ref.update(updated, function() {
+            ctrl.saving = false;
+            m.redraw();
         });
     };
 
     ctrl.unpublish = function() {
-        ref.update({
-            // TODO: Remove `published` field, it's deprecated
-            published    : null,
-            published_at : null,
-            published_by : null,
+        var nowTs,
+            updated;
 
-            unpublished_at : db.TIMESTAMP,
-            unpublished_by : user
-        });
+        if(ctrl.end.ts && isPast(ctrl.end.ts)) {
+            // Already unpublished.
+            return;
+        }
 
-        // TODO: THIS IS SUPER GROSS
-        ctrl.start = {
-            date : "",
-            time : ""
-        };
+        ctrl.saving = true;
+        m.redraw();
 
-        ctrl.end = {
-            date : "",
-            time : ""
+        nowTs = Date.now();
+        ctrl.end = ctrl.makeScheduleObj(nowTs);
+
+        updated = {
+            unpublished_at : nowTs,
+            unpublished_by : user
         };
+
+        if(ctrl.start.ts &&  ctrl.start.ts > ctrl.end.ts) {
+            // Invalid start.
+            ctrl.start = ctrl.nulledScheduleStart();
+        }
+
+        ref.update(updated, function() {
+            ctrl.saving = false;
+            m.redraw();
+        });
+    };
+
+    ctrl.clearSchedule = function() {
+        ctrl.start = ctrl.nulledScheduleStart();
+        ctrl.end   = ctrl.nulledScheduleEnd();
+        ctrl.invalidDates = false;
     };
 
     ctrl.save = function(opts) {
-        ctrl.saving = true;
+        var updated = {};
 
+        ctrl.saving = true;
         m.redraw();
 
-        ref.update({
+        updated = {
             fields : opts.data.fields,
             name   : opts.data.name,
             slug   : opts.data.slug || null
-        }, function() {
-            ctrl.saving = false;
+        };
+
+        updated = ctrl.addScheduleData(updated);
 
+        ref.update(updated, function() {
+            ctrl.saving = false;
             m.redraw();
         });
     };
+
+    ctrl.addScheduleData = function(updated) {
+        if(!ctrl.invalidDates) {
+            let startTs = null,
```

Comment:
Can't remember exactly why this function ended up so weird. I think it was because I couldn't set vars after the early-out check and that bugged me, and I think I just ended up fighting with the style linter until it ended up like this.